File: SEMESTRE 2/PROG2/Ejercicios de DiagramaCompletos/PracticaDiagramas-Barreto-Matias/PracticaDiagramas-Barreto-Matias/src/ejerciciodos/Detalle_Factura.java
package ejerciciodos;

/**
 *
 * @author Matias Barreto Legajo: 49215 mail: barretomatias93@gmail.com
 */
public class Detalle_Factura {

    private int cod_factura;
    private int cod_articulo;
    private int cantidad;
    private int IVA;
    private int precioVenta;
    private int descuento;

    Factura factura = new Factura();
    Articulo articulo = new Articulo();

    public int getCod_factura() {
        return cod_factura;
    }

    public void setCod_factura(int cod_factura) {
        this.cod_factura = cod_factura;
    }

    public int getCod_articulo() {
        return cod_articulo;
    }

    public void setCod_articulo(int cod_articulo) {
        this.cod_articulo = cod_articulo;
    }

    public int getCantidad() {
        return cantidad;
    }

    public void setCantidad(int cantidad) {
        this.cantidad = cantidad;
    }

    public int getIVA() {
        return IVA;
    }

    public void setIVA(int IVA) {
        this.IVA = IVA;
    }

    public int getPrecioVenta() {
        return precioVenta;
    }

    public void setPrecioVenta(int precioVenta) {
        this.precioVenta = precioVenta;
    }

    public int getDescuento() {
        return descuento;
    }

    public void setDescuento(int descuento) {
        this.descuento = descuento;
    }    
}

File: SEMESTRE 3/LAB3/TP7-TS-ModeloB/DetalleFactura.ts
import { Factura } from "./Factura";
import { Articulo } from "./Articulo";

export class DetalleFactura {
    constructor(cantidad: number, articulo: Articulo) {
        this._cantidad = cantidad;
        this._articulo = articulo;
    }

    private _cantidad: number; 
    public get cantidad(): number {
        return this._cantidad;
    }
    public set cantidad(value: number) {
        this._cantidad = value;
    }

    private _subtotal: number;
    public get subtotal(): number {
        return this._subtotal;
    }
    public set subtotal(value: number) {
        this._subtotal = value;
    }

    private _factura: Factura;
    public get factura(): Factura {
        return this._factura;
    }
    public set factura(value: Factura) {
        this._factura = value;
    }

    private _articulo: Articulo; 
    public get articulo(): Articulo {
        return this._articulo;
    }
    public set articulo(value: Articulo) {
        this._articulo = value;
    }

    public calcularSubTotal(){
        this._subtotal = this._cantidad * this._articulo.precio;
    }

}